.Rprofile

```diff
@@ -1,65 +1,21 @@
+source("renv/activate.R")
+
+# bootstrapping, will only run once after remind is freshly cloned
+if (nrow(installed.packages(priority = "NA")) == 1) {
+  # only one package is installed, presumably renv
+  message("R package dependencies are not installed in this renv, installing now...")
+  renv::install("yaml", prompt = FALSE) # yaml is required to find dependencies in Rmd files
+  renv::hydrate() # auto-detect and install all dependencies
+  renv::snapshot(prompt = FALSE) # create renv.lock
+  message("Finished installing R package dependencies.")
+}
+
 # source global .Rprofile (very important to load user specific settings)
 # DO NOT EDIT THIS LINE!
 if (file.exists("~/.Rprofile")) {
   source("~/.Rprofile")
 }
 
-# This profile can be used to link the model to a specified library snapshot
-# (e.g. if your model version is from an older date and does not work with the
-# newest libraries anymore). By default it is not active.
-
-local({ # prevent variables defined here from ending up in the global env
-
-# snapshots created with R 4.1 have a _R4 postfix
-maybeR4 <- if (R.version$major == 4) "_R4" else ""
-# Gather all library snapshots
-snapshots <- sort(list.files(path = '/p/projects/rd3mod/R/libraries/snapshots',
-                             pattern = paste0('^20[0-9]{2}_[0-9]{2}(_[0-9]{2})?', maybeR4, '$'),
-                             full.names = TRUE), decreasing = TRUE)
-latestSnapshot <- if (length(snapshots) > 0) snapshots[[1]] else NA
-
-activateSnapshot <- function(snapshot) {
-  stopifnot(file.exists(snapshot))
-  if (R.version$major <= 3) { # include.site is not available before R 4.0
-    if (endsWith(snapshot, "_R4")) stop("Your R version is ", R.version$major, ", but your library snapshot is for 4.0 or later")
-    .libPaths(snapshot)
-  } else {
-    if (!endsWith(snapshot, "_R4")) stop("Your R version is ", R.version$major, ", but your library snapshot is for < 4.0.")
-    # setting include.site to FALSE makes sure that only the snapshot and system libraries are used
-    .libPaths(snapshot, include.site = FALSE)
-  }
-  message("libPaths was set to: ", snapshot)
-}
-
-# Just uncomment the following line and set the snapshot path
-# to a path of your choice or use the second line to use the latest snapshot.
-# Please make also sure that in your config file this .Rprofile file is copied
-# to the model output folder. Otherwise, the run itself will again use the
-# default library set!
-# Snapshots must be compatible to the R version used. If you are using R 4.1
-# make sure the selected snapshot's name ends with '_R4'.
-
-# snapshot <- "/p/projects/rd3mod/R/libraries/snapshots/2022_05_31_R4"
-# snapshot <- latestSnapshot
-
-if (exists("snapshot")) {
-  activateSnapshot(snapshot)
+if (!"https://rse.pik-potsdam.de/r/packages" %in% getOption("repos")) {
+  options(repos = c(getOption("repos"), pik = "https://rse.pik-potsdam.de/r/packages"))
```

Comment:
The auto-setup will fail without it, because it cannot find all the PIK packages. People from PIK will have this configured probably, so it's just a fallback really, but for people outside PIK it should help.

Reply:
See comment on `1_GettinREMIND.md`.  This has to go before the bootstrapping, else new user (everybody that does not have the required packages installed) will be left with a defunct REMIND installation.